### PR TITLE
refactor: new image parsing on `CustomizedImageList`

### PR DESCRIPTION
**Pull request by agatha197**
**Changes:**
Adds search functionality and enhanced display features to the customized image list, including:
- New search bar with real-time filtering
- Refresh button to reload image data
- Copy button for full image paths
- Text highlighting for search matches
- New columns for base image name, version, and tags when extended image info is supported
- Improved table loading states using transitions

**Minimum required manager version:** 24.09.1

**How to test:**
> endpoint: 10.100.64.15
1. Checkout core branch to `topic/10-22-feature_add_info_field_to_gql_image_schema`
2. Visit My Environments page.

**What to check:**
- [ ] Check the data is valid.
  - [ ]  Since 24.09.1: Full image path, Registry, Architecture, Namespace, Base image name, Version, Tags, Digest, Control.
  - [ ] Before 24.09.1: Full image path, Registry, Architecture, Name, Language, Version, Base, Constraints, Digest, Control.
- [ ] All data is highlightable by searching images.
- [ ] Sortable data works fine.

**Requirements to check during review:**
- Search functionality filters across all relevant fields
- Text highlighting properly emphasizes search terms
- Copy button correctly copies the full image path
- Table displays appropriate columns based on extended image info support
- Loading states show correctly during search and refresh operations

The changes improve the user experience by making image information more accessible and searchable while maintaining backwards compatibility with existing functionality.

### CODE REVIEW

**Comment by agatha197** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2795?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2795** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2795?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2794** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2794?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2788** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2788?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2789** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2789?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2787** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2787?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2785** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2785?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @agatha197 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/CustomizedImageList.tsx [hunk 1]
```diff
@@ -45,6 +54,8 @@ const CustomizedImageList: React.FC<PropsWithChildren> = ({ children }) => {
   const [customizedImageListFetchKey, updateCustomizedImageListFetchKey] =
     useUpdatableState('initial-fetch');
   const [inFlightImageId, setInFlightImageId] = useState<string>();
+  const [imageSearch, setImageSearch] = useState('');
```
Comment (ironAiken2) [inline react/src/components/CustomizedImageList.tsx:59]: It looks like you need imageSearch string based item filtering. 

[ref](https://github.com/lablup/backend.ai-webui/blob/188d31f8bd994635ee8e24dfc108663041a5d442/react/src/components/ImageList.tsx#L359-L394)

File: react/src/components/CustomizedImageList.tsx [hunk 2]
```diff
@@ -101,17 +120,42 @@ const CustomizedImageList: React.FC<PropsWithChildren> = ({ children }) => {
...
 
   const columns: ColumnsType<CommittedImage> = [
+    {
+      title: t('environment.FullImagePath'),
+      key: 'fullImagePath',
+      render: (row) => (
+        <Flex gap={'xxs'}>
+          <TextHighlighter keyword={imageSearch}>
+            {getImageFullName(row) || ''}
+          </TextHighlighter>
```
Comment (ironAiken2) [inline react/src/components/CustomizedImageList.tsx:223]: Commented on [Previous Stack](https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2794/refactor-MyEnvironmentPage-into-two-components#comment-PRRC_kwDOCRTcws5szp4Z) about using the CopyButton component and the criteria for using copyable in Typography.Text.

File: react/src/components/CustomizedImageList.tsx [hunk 3]
```diff
@@ -139,100 +224,96 @@ const CustomizedImageList: React.FC<PropsWithChildren> = ({ children }) => {
               <span>{getNamespace(getImageFullName(row) || '')}</span>
             ),
           },
+          {
+            title: t('environment.Language'),
+            key: 'lang',
+            sorter: (a: CommittedImage, b: CommittedImage) => {
+              const langA = getImageLang(getImageFullName(a) || '');
+              const langB = getImageLang(getImageFullName(b) || '');
+              return langA && langB ? langA.localeCompare(langB) : 0;
```
Comment (ironAiken2) [inline react/src/components/CustomizedImageList.tsx]:
```suggestion
              return localeCompare(langA, langB);
```

File: react/src/components/CustomizedImageList.tsx [hunk 4]
```diff
@@ -139,100 +224,96 @@ const CustomizedImageList: React.FC<PropsWithChildren> = ({ children }) => {
...
+                b?.tag && b?.labels
+                  ? getConstraints(
+                      b?.tag,
+                      b?.labels as { key: string; value: string }[],
+                    )[0] || ''
+                  : '';
+              if (requirementA === '' && requirementB === '') return 0;
+              if (requirementA === '') return -1;
+              if (requirementB === '') return 1;
+              return requirementA.localeCompare(requirementB);
```
Comment (ironAiken2) [inline react/src/components/CustomizedImageList.tsx]:
```suggestion
              return localeCompare(requirementA, requirementB);
```

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
I left some review about sorting and not being able to filter items based on search string.

File: react/src/components/CustomizedImageList.tsx (hunk 2 above)
Comment (agatha197) [reply react/src/components/CustomizedImageList.tsx:223]: I was actually going to make a new thread to discuss about that :)

**Review by yomybaby — APPROVED** [review]
LGTM

**Review by yomybaby — APPROVED** [review]
LGTM

**Pull request merged**